utility.lua

```diff
@@ -129,25 +129,39 @@ hopper.take_item_from = function(hopper_pos, target_pos, target_node, target_inv
 	end
 end
 
-local function send_item_to_air(hopper_inv, target_pos, filtered_items)
+local function send_item_to_air(hopper_inv, target_pos)
+	local item
 	local stack
 	local stack_num
 	for i = 1, hopper_inv:get_size("main") do
 		stack = hopper_inv:get_stack("main", i)
-		local item = stack:get_name()
-		if item ~= "" and (filtered_items == nil or filtered_items[item]) then
+		item = stack:get_name()
+		if item ~= "" then
 			stack_num = i
 			break
 		end
 	end
-	if not stack_num then
-		return false
+
+	local eject_node = minetest.get_node(target_pos)
+	if (eject_node["name"] ~= "air") then
```

Comment:
Ejecting to any non-solids should be fine. This also closely follows the now deleted code in `hopper.send_item_to`.

```Lua
local ndef = minetest.registered_nodes[eject_node.name]
if not ndef or not ndef.buildable_to then
```

Reply:
Good thinking. I may have replaced it after doing the find node for air...